raiden/tests/mocked/test_matrix_transport.py

```diff
@@ -0,0 +1,160 @@
+from raiden.utils.signer import LocalSigner
+
+USERID0 = "@0x1234567890123456789012345678901234567890:RestaurantAtTheEndOfTheUniverse"
+USERID1 = "@0x0987654321098765432109876543210987654321:Wonderland"
+
+
+@pytest.fixture()
+def skip_userid_validation(monkeypatch):
+    import raiden.network.transport.matrix
+    import raiden.network.transport.matrix.utils
+
+    def mock_validate_userid_signature(user):  # pylint: disable=unused-argument
+        return factories.HOP1
+
+    monkeypatch.setattr(
+        raiden.network.transport.matrix,
+        "validate_userid_signature",
+        mock_validate_userid_signature,
+    )
+    monkeypatch.setattr(
+        raiden.network.transport.matrix.transport,
+        "validate_userid_signature",
+        mock_validate_userid_signature,
+    )
+    monkeypatch.setattr(
+        raiden.network.transport.matrix.utils,
+        "validate_userid_signature",
+        mock_validate_userid_signature,
+    )
+
+
+@pytest.fixture
+def mock_matrix(monkeypatch, retry_interval, retries_before_backoff):
+
+    from raiden.network.transport.matrix.client import User
+    from raiden.network.transport.matrix import transport
+
+    monkeypatch.setattr(User, "get_display_name", lambda _: "random_display_name")
+    monkeypatch.setattr(transport, "make_client", lambda url, *a, **kw: GMatrixClient(url[0]))
+
+    def mock_get_user(klass, user: Union[User, str]) -> User:  # pylint: disable=unused-argument
+        return User(None, USERID1)
+
+    def mock_get_room_ids_for_address(  # pylint: disable=unused-argument
+        klass, address: Address, filter_private: bool = None
+    ) -> List[str]:
+        return ["!roomID:server"]
+
+    def mock_set_room_id_for_address(  # pylint: disable=unused-argument
+        self, address: Address, room_id: Optional[str]
+    ):
+        pass
+
+    def mock_receive_message(klass, message):  # pylint: disable=unused-argument
+        # We are just unit testing the matrix transport receive so do nothing
+        assert message
+
+    config = dict(
+        retry_interval=retry_interval,
+        retries_before_backoff=retries_before_backoff,
+        server="http://none",
+        server_name="none",
+        available_servers=[],
+        global_rooms=[],
+        private_rooms=False,
+    )
+
+    transport = MatrixTransport(config)
+    transport._raiden_service = MockRaidenService()
+    transport._stop_event.clear()
+    transport._address_mgr.add_userid_for_address(factories.HOP1, USERID1)
+    transport._client.user_id = USERID0
+
+    monkeypatch.setattr(MatrixTransport, "_get_user", mock_get_user)
+    monkeypatch.setattr(
+        MatrixTransport, "_get_room_ids_for_address", mock_get_room_ids_for_address
+    )
+    monkeypatch.setattr(MatrixTransport, "_set_room_id_for_address", mock_set_room_id_for_address)
+    monkeypatch.setattr(MatrixTransport, "_receive_message", mock_receive_message)
+
+    return transport
+
+
+def make_message(overwrite_data=None):
```

Comment:
nit: type annotations

Reply:
Aren't the tests still excluded from typing? 
Adding types here makes mypy throw a whole bunch of annoying errors because of mocking.

Reply:
Yes, if it's to much work just skip adding them